PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/test/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ConnectionStringCheckTest.java

```diff
@@ -0,0 +1,133 @@
+
+        assertVisitor(mockVisitor);
+        String METHOD_TO_CHECK = "connectionString";
+        int NUMBER_OF_INVOCATIONS = 1;
+        String CLASS_NAME = "com.azure";
+
+        verifyRegisterProblem(mockVisitor, METHOD_TO_CHECK, NUMBER_OF_INVOCATIONS, CLASS_NAME);
+    }
+
+    // Problem isn't registered because the method to check is different from the method that should be flagged
+    @Test
+    void differentMethodCheck() {
+        String METHOD_TO_CHECK = "differentMethod";
+        int NUMBER_OF_INVOCATIONS = 0;
+        String CLASS_NAME = "com.azure";
+
+        verifyRegisterProblem(mockVisitor, METHOD_TO_CHECK, NUMBER_OF_INVOCATIONS, CLASS_NAME);
+    }
+
+    // Problem isn't registered because the class name is different from the class that should be flagged
+    @Test
+    void differentClassCheck() {
+        String METHOD_TO_CHECK = "connectionString";
+        int NUMBER_OF_INVOCATIONS = 0;
+        String CLASS_NAME = "com.microsoft.azure";
+
+        verifyRegisterProblem(mockVisitor, METHOD_TO_CHECK, NUMBER_OF_INVOCATIONS, CLASS_NAME);
+    }
+
+    // Problem isn't registered because the class name is null
+    @Test
+    void nullClassCheck() {
+        String METHOD_TO_CHECK = "connectionString";
+        int NUMBER_OF_INVOCATIONS = 0;
+        String CLASS_NAME = null;
+
+        verifyRegisterProblem(mockVisitor, METHOD_TO_CHECK, NUMBER_OF_INVOCATIONS, CLASS_NAME);
+    }
+
+    // Asserts that the visitor is not null and is an instance of JavaElementVisitor
+    void assertVisitor(PsiElementVisitor visitor) {
+        assertNotNull(visitor);
+        assertTrue(visitor instanceof JavaElementVisitor);
+    }
+
+    // Verifies that a warning is raised when a connection string being used to create a client is detected.
+    void verifyRegisterProblem(PsiElementVisitor visitor, String METHOD_TO_CHECK, int NUMBER_OF_INVOCATIONS, String CLASS_NAME) {
+
+        PsiMethodCallExpression methodCallExpression = mock(PsiMethodCallExpression.class);
+        PsiReferenceExpression methodExpression = mock(PsiReferenceExpression.class);
+        PsiMethod resolvedMethod = mock(PsiMethod.class);
+        PsiMethod method = mock(PsiMethod.class);
+        PsiClass containingClass = mock(PsiClass.class);
+
+        when(methodCallExpression.getMethodExpression()).thenReturn(methodExpression);
+        when(methodExpression.resolve()).thenReturn(resolvedMethod);
+        when(resolvedMethod.getContainingClass()).thenReturn(containingClass);
+        when(resolvedMethod.getName()).thenReturn(METHOD_TO_CHECK);
+        when(containingClass.getQualifiedName()).thenReturn(CLASS_NAME);
+
+        (visitor).visitElement(methodCallExpression);
+
+        // Verify problem is registered
+        verify(mockHolder, times(NUMBER_OF_INVOCATIONS)).registerProblem(Mockito.eq(methodCallExpression), Mockito.contains("Connection String detected. Use AzureKeyCredentials for azure service client authentication instead."));
```

Comment:
This should be failing, the rule configured suggests to use `DefaultAzureCredential` and not AzureKeyCredentials.

Reply:
yes hehe I made the changes and it passes now